lib/ecto_crux.ex

```diff
@@ -102,6 +102,8 @@ defmodule EctoCrux do
       @page_size args[:page_size] ||
                    Application.get_all_env(:ecto_crux)[:page_size] ||
                    50
+      # allow to not defined functions that are not defined when using Repo with a read_only mode
+      @read_only args[:read_only] || false
```

Comment:
I'm really not ok with this. Don't hide your fear in magic config !
If you really want to be able to select what function to import, you may implement like an import:
```
      use EctoCrux, module: EctoCrux.Schema.Baguette, only: [:get, :find_by]
```

Reply:
If I may add some context, this is done to be compatible with https://hexdocs.pm/ecto/Ecto.Repo.html#module-read-only-repositories.

Reply:
Thanks to @creusefond, that's why i implemented it like that. If a read_only Repo is set, all the function that are using the underlying repo functions `insert`, `update `or `delete `(write operation) will raise a warning since those function would not be defined. 
I made that to follow Repo, so we may call the flag `read_only_repo` to be precise ? In order to avoid what you are saying

More over, tomorrow if we are adding a bunch of functions in crux that use underlying Repo read functions. It would be, in my opinion, better to not modify the `use` like you propose when updating crux to just add those hypothetic functions.

Reply:
TIL.

Ok for me then !